src/Reflection/ReflectionClass.php

```diff
@@ -543,6 +536,45 @@ function (ReflectionClassConstant $constant) : string {
     }
 
     /**
+     * Get an associative array of the defined constants in this class,
+     * with keys as constant names and values as {@see ReflectionClassConstant} objects.
+     *
+     * @return ReflectionClassConstant[]
+     */
+    public function getReflectionConstants() : array
+    {
+        $allReflectionConstants = array_merge(
```

Comment:
Note: I tried simplifying this code by removing the merging oddities:

```php
return array_merge(
            ...array_map(
                function (ReflectionClass $ancestor) {
                    return array_filter(
                        $ancestor->getReflectionConstants(),
                        function (ReflectionClassConstant $classConstant) {
                            return ! $classConstant->isPrivate();
                        }
                    );
                },
                array_filter([$this->getParentClass()])
            ),
            ...array_map(
                function (ReflectionClass $interface) {
                    return $interface->getReflectionConstants();
                },
                array_values($this->getInterfaces())
            ),
            ...[$this->getImmediateReflectionConstants()]
        );
```

The sorting does not respect the correct expectations though:

```php
<?php

interface Foo {
    const A = 'a';
}

interface Boo {
    public const B = 'b';
}

class Baz implements Foo {
    public const C = 'c';

    protected const D = 'd';

    private const E = 'e';
}

abstract class Qux extends Baz implements Boo {
    public const F = 'f';

    protected const D = 'dd';
}

abstract class Next extends Qux {
    public const F = 'ff';
}

var_dump((new ReflectionClass(Next::class))->getConstants());
```

Which produces following output, [according to 3v4l.org](https://3v4l.org/oTheI):

```
array(5) { ["F"]=> string(2) "ff" ["D"]=> string(2) "dd" ["C"]=> string(1) "c" ["A"]=> string(1) "a" ["B"]=> string(1) "b" }
```

Therefore this can't be simplified further, I fear :-(